### ISSUE TITLE
BUG in fs/bcachefs/btree_cache.c [cb8869d]

### ISSUE THREAD

**Issue by MoraFermi** [koverstreet/bcachefs#466]
This is freshly compiled cb8869d : bcachefs: Fix a deadlock in btree_update_nodes_written()

This one happened during normal FS usage. FS was mounted with `fix_errors,fsck`, so it should have been clean.

```
[Tue Oct  4 10:20:51 2022] ------------[ cut here ]------------
[Tue Oct  4 10:20:51 2022] kernel BUG at fs/bcachefs/btree_cache.c:630!
[Tue Oct  4 10:20:51 2022] invalid opcode: 0000 [#1] PREEMPT SMP PTI
[Tue Oct  4 10:20:51 2022] CPU: 1 PID: 1794 Comm: bch-reclaim/b75 Not tainted 6.0.0-t14 #1
[Tue Oct  4 10:20:51 2022] Hardware name: HP HP EliteDesk 800 G3 DM 65W/829A, BIOS P21 Ver. 02.39 11/22/2021
[Tue Oct  4 10:20:51 2022] RIP: 0010:bch2_btree_node_mem_alloc+0x1c2/0x5d0
[Tue Oct  4 10:20:51 2022] Code: 01 00 00 4c 89 ab 50 10 00 00 4c 89 f7 e8 a6 1f bd ff 4c 89 f7 e8 ee 21 bd ff 48 8b 7d d0 e8 85 31 9e 00 49 83 7f 50 00 74 46 <0f> 0b be 00 20 00 00 48 89 df e8 ff e3 ff ff 49 89 c7 48 85 c0 0f
[Tue Oct  4 10:20:51 2022] RSP: 0000:ffffbb1700b075d0 EFLAGS: 00010286
[Tue Oct  4 10:20:51 2022] RAX: ffff939210a34e00 RBX: ffff939207cc0000 RCX: 0000000000000004
[Tue Oct  4 10:20:51 2022] RDX: 0000000000000000 RSI: 0000000000000000 RDI: ffff939207cc1010
[Tue Oct  4 10:20:51 2022] RBP: ffffbb1700b07610 R08: 0000000000000001 R09: 0000000000000000
[Tue Oct  4 10:20:51 2022] R10: 0000000000000004 R11: fffffffffffffff8 R12: ffff939207cc1040
[Tue Oct  4 10:20:51 2022] R13: ffff939204f1d3d0 R14: ffff939204f1d200 R15: ffff9392ffa22c00
[Tue Oct  4 10:20:51 2022] FS:  0000000000000000(0000) GS:ffff93990f680000(0000) knlGS:0000000000000000
[Tue Oct  4 10:20:51 2022] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[Tue Oct  4 10:20:51 2022] CR2: 00007fe735ca0070 CR3: 00000006d94de004 CR4: 00000000003706e0
[Tue Oct  4 10:20:51 2022] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
[Tue Oct  4 10:20:51 2022] DR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400
[Tue Oct  4 10:20:51 2022] Call Trace:
[Tue Oct  4 10:20:51 2022]  <TASK>
[Tue Oct  4 10:20:51 2022]  __bch2_btree_node_alloc+0x16e/0x4f0
[Tue Oct  4 10:20:51 2022]  bch2_btree_reserve_get+0xdb/0x150
[Tue Oct  4 10:20:51 2022]  bch2_btree_update_start+0x5e5/0x7d0
[Tue Oct  4 10:20:51 2022]  __bch2_foreground_maybe_merge+0x5a2/0x1140
[Tue Oct  4 10:20:51 2022]  ? bch2_btree_path_peek_slot+0xaf/0x320
[Tue Oct  4 10:20:51 2022]  __bch2_trans_commit+0x1ae5/0x2060
[Tue Oct  4 10:20:51 2022]  ? btree_key_cache_flush_pos+0x4c5/0x6a0
[Tue Oct  4 10:20:51 2022]  btree_key_cache_flush_pos+0x4c5/0x6a0
[Tue Oct  4 10:20:51 2022]  ? find_busiest_group+0x48/0x350
[Tue Oct  4 10:20:51 2022]  ? btree_key_cache_flush_pos+0xa1/0x6a0
[Tue Oct  4 10:20:51 2022]  ? bch2_btree_key_cache_journal_flush+0x100/0x1f0
[Tue Oct  4 10:20:51 2022]  ? btree_key_cache_flush_pos+0x6e/0x6a0
[Tue Oct  4 10:20:51 2022]  bch2_btree_key_cache_journal_flush+0x12b/0x1f0
[Tue Oct  4 10:20:51 2022]  ? bch2_btree_key_cache_journal_flush+0x100/0x1f0
[Tue Oct  4 10:20:51 2022]  ? btree_key_cache_flush_pos+0x6a0/0x6a0
[Tue Oct  4 10:20:51 2022]  journal_flush_pins+0x133/0x250
[Tue Oct  4 10:20:51 2022]  __bch2_journal_reclaim+0x2a4/0x4a0
[Tue Oct  4 10:20:51 2022]  bch2_journal_reclaim_thread+0x7f/0x1e0
[Tue Oct  4 10:20:51 2022]  ? __bch2_journal_reclaim+0x4a0/0x4a0
[Tue Oct  4 10:20:51 2022]  kthread+0xfa/0x130
[Tue Oct  4 10:20:51 2022]  ? kthread_complete_and_exit+0x20/0x20
[Tue Oct  4 10:20:51 2022]  ret_from_fork+0x1f/0x30
[Tue Oct  4 10:20:51 2022]  </TASK>
[Tue Oct  4 10:20:51 2022] Modules linked in: ebtable_filter ebtables ip_set ip6table_raw iptable_raw ip6table_filter ip6_tables iptable_filter sctp ip6_udp_tunnel udp_tunnel nf_tables 8021q garp mrp softdog bonding tls openvswitch nsh nf_conncount nf_nat nf_conntrack nf_defrag_ipv6 nf_defrag_ipv4 nfnetlink_log nfnetlink snd_hda_codec_hdmi snd_hda_codec_conexant snd_hda_codec_generic ledtrig_audio intel_rapl_msr intel_rapl_common intel_tcc_cooling x86_pkg_temp_thermal intel_powerclamp coretemp kvm_intel kvm irqbypass iwlmvm crct10dif_pclmul mei_hdcp ghash_clmulni_intel i915 mac80211 libarc4 aesni_intel crypto_simd cryptd snd_hda_intel drm_buddy rapl intel_cstate btusb snd_intel_dspcfg ttm snd_intel_sdw_acpi iwlwifi btrtl drm_display_helper snd_hda_codec btbcm cec hp_wmi btintel pcspkr platform_profile btmtk cdc_mbim rc_core snd_hda_core cdc_wdm snd_hwdep wmi_bmof sparse_keymap efi_pstore snd_pcm drm_kms_helper i2c_algo_bit ee1004 fb_sys_fops cfg80211 syscopyarea bluetooth snd_timer mei_me cdc_ncm snd
[Tue Oct  4 10:20:51 2022]  sysfillrect cdc_ether ecdh_generic ecc usbnet soundcore sysimgblt mii mei acpi_pad mac_hid vhost_net vhost vhost_iotlb tap ib_iser rdma_cm iw_cm ib_cm ib_core iscsi_tcp libiscsi_tcp libiscsi scsi_transport_iscsi drm sunrpc ip_tables x_tables autofs4 btrfs blake2b_generic simplefb uas usb_storage nvme i2c_i801 crc32_pclmul i2c_smbus e1000e xhci_pci xhci_pci_renesas ahci nvme_core libahci xhci_hcd wmi video
[Tue Oct  4 10:20:51 2022] ---[ end trace 0000000000000000 ]---
[Tue Oct  4 10:20:51 2022] RIP: 0010:bch2_btree_node_mem_alloc+0x1c2/0x5d0
[Tue Oct  4 10:20:51 2022] Code: 01 00 00 4c 89 ab 50 10 00 00 4c 89 f7 e8 a6 1f bd ff 4c 89 f7 e8 ee 21 bd ff 48 8b 7d d0 e8 85 31 9e 00 49 83 7f 50 00 74 46 <0f> 0b be 00 20 00 00 48 89 df e8 ff e3 ff ff 49 89 c7 48 85 c0 0f
[Tue Oct  4 10:20:51 2022] RSP: 0000:ffffbb1700b075d0 EFLAGS: 00010286
[Tue Oct  4 10:20:51 2022] RAX: ffff939210a34e00 RBX: ffff939207cc0000 RCX: 0000000000000004
[Tue Oct  4 10:20:51 2022] RDX: 0000000000000000 RSI: 0000000000000000 RDI: ffff939207cc1010
[Tue Oct  4 10:20:51 2022] RBP: ffffbb1700b07610 R08: 0000000000000001 R09: 0000000000000000
[Tue Oct  4 10:20:51 2022] R10: 0000000000000004 R11: fffffffffffffff8 R12: ffff939207cc1040
[Tue Oct  4 10:20:51 2022] R13: ffff939204f1d3d0 R14: ffff939204f1d200 R15: ffff9392ffa22c00
[Tue Oct  4 10:20:51 2022] FS:  0000000000000000(0000) GS:ffff93990f680000(0000) knlGS:0000000000000000
[Tue Oct  4 10:20:51 2022] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[Tue Oct  4 10:20:51 2022] CR2: 00007fe735ca0070 CR3: 00000006d94de004 CR4: 00000000003706e0
[Tue Oct  4 10:20:51 2022] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
[Tue Oct  4 10:20:51 2022] DR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400
[Tue Oct  4 10:20:51 2022] BUG: kernel NULL pointer dereference, address: 0000000000000016
[Tue Oct  4 10:20:51 2022] #PF: supervisor write access in kernel mode
[Tue Oct  4 10:20:51 2022] #PF: error_code(0x0002) - not-present page
[Tue Oct  4 10:20:51 2022] PGD 0 P4D 0
[Tue Oct  4 10:20:51 2022] Oops: 0002 [#2] PREEMPT SMP PTI
[Tue Oct  4 10:20:51 2022] CPU: 3 PID: 30302 Comm: smbd Tainted: G      D            6.0.0-t14 #1
[Tue Oct  4 10:20:51 2022] Hardware name: HP HP EliteDesk 800 G3 DM 65W/829A, BIOS P21 Ver. 02.39 11/22/2021
[Tue Oct  4 10:20:51 2022] RIP: 0010:__bch2_trans_init+0x25b/0x340
[Tue Oct  4 10:20:51 2022] Code: 40 48 8d 50 c0 48 39 f0 74 3e 48 8b 4a 78 3b b9 e0 09 00 00 7d e7 48 8b 4a 48 49 8d 76 40 48 89 72 48 49 89 46 40 49 89 4e 48 <48> 89 31 4c 89 e7 e8 ea 39 9c 00 48 83 c4 10 5b 41 5c 41 5d 41 5e
[Tue Oct  4 10:20:51 2022] RSP: 0018:ffffbb1726443a10 EFLAGS: 00010283
[Tue Oct  4 10:20:51 2022] RAX: ffffbb1700b07ca0 RBX: 00000000000026f8 RCX: 0000000000000016
[Tue Oct  4 10:20:51 2022] RDX: ffffbb1700b07c60 RSI: ffffbb1726443aa0 RDI: 000000000000765e
[Tue Oct  4 10:20:51 2022] RBP: ffffbb1726443a48 R08: 0000000000000000 R09: 0000000000000008
[Tue Oct  4 10:20:51 2022] R10: ffffffff858787a0 R11: 0000000000000000 R12: ffff939207cc3168
[Tue Oct  4 10:20:51 2022] R13: 000000000000001d R14: ffffbb1726443a60 R15: ffff939207cc0000
[Tue Oct  4 10:20:51 2022] FS:  00007fe7318d6a40(0000) GS:ffff93990f780000(0000) knlGS:0000000000000000
[Tue Oct  4 10:20:51 2022] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[Tue Oct  4 10:20:51 2022] CR2: 0000000000000016 CR3: 00000006d94de003 CR4: 00000000003706e0
[Tue Oct  4 10:20:51 2022] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
[Tue Oct  4 10:20:51 2022] DR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400
[Tue Oct  4 10:20:51 2022] Call Trace:
[Tue Oct  4 10:20:51 2022]  <TASK>
[Tue Oct  4 10:20:51 2022]  bch2_xattr_get+0x4d/0x140
[Tue Oct  4 10:20:51 2022]  bch2_xattr_get_handler+0x2c/0x40
[Tue Oct  4 10:20:51 2022]  vfs_getxattr+0x17f/0x210
[Tue Oct  4 10:20:51 2022]  do_getxattr+0x6b/0x160
[Tue Oct  4 10:20:51 2022]  getxattr+0x92/0xd0
[Tue Oct  4 10:20:51 2022]  ? memcg_slab_post_alloc_hook+0x1b8/0x260
[Tue Oct  4 10:20:51 2022]  ? debug_smp_processor_id+0x17/0x20
[Tue Oct  4 10:20:51 2022]  ? proc_nr_files+0x30/0x30
[Tue Oct  4 10:20:51 2022]  ? debug_smp_processor_id+0x17/0x20
[Tue Oct  4 10:20:51 2022]  ? call_rcu+0x128/0x260
[Tue Oct  4 10:20:51 2022]  ? __fget_light+0xa7/0x130
[Tue Oct  4 10:20:51 2022]  __x64_sys_fgetxattr+0x6b/0xc0
[Tue Oct  4 10:20:51 2022]  do_syscall_64+0x59/0x90
[Tue Oct  4 10:20:51 2022]  ? exit_to_user_mode_prepare+0x49/0x1a0
[Tue Oct  4 10:20:51 2022]  ? syscall_exit_to_user_mode+0x26/0x50
[Tue Oct  4 10:20:51 2022]  ? __x64_sys_close+0x11/0x50
[Tue Oct  4 10:20:51 2022]  ? do_syscall_64+0x69/0x90
[Tue Oct  4 10:20:51 2022]  ? fpregs_assert_state_consistent+0x2a/0x50
[Tue Oct  4 10:20:51 2022]  ? exit_to_user_mode_prepare+0x49/0x1a0
[Tue Oct  4 10:20:51 2022]  ? syscall_exit_to_user_mode+0x26/0x50
[Tue Oct  4 10:20:51 2022]  ? __do_sys_getpid+0x1e/0x30
[Tue Oct  4 10:20:51 2022]  ? do_syscall_64+0x69/0x90
[Tue Oct  4 10:20:51 2022]  ? do_syscall_64+0x69/0x90
[Tue Oct  4 10:20:51 2022]  ? do_syscall_64+0x69/0x90
[Tue Oct  4 10:20:51 2022]  ? do_syscall_64+0x69/0x90
[Tue Oct  4 10:20:51 2022]  entry_SYSCALL_64_after_hwframe+0x63/0xcd
[Tue Oct  4 10:20:51 2022] RIP: 0033:0x7fe73575547a
[Tue Oct  4 10:20:51 2022] Code: 48 8b 05 09 5a 0d 00 ba 64 00 00 00 8b 40 38 85 c0 0f 44 c2 c3 66 2e 0f 1f 84 00 00 00 00 00 90 49 89 ca b8 c1 00 00 00 0f 05 <48> 3d 01 f0 ff ff 73 01 c3 48 8b 0d e6 59 0d 00 f7 d8 64 89 01 48
[Tue Oct  4 10:20:51 2022] RSP: 002b:00007fffa420f9d8 EFLAGS: 00000202 ORIG_RAX: 00000000000000c1
[Tue Oct  4 10:20:51 2022] RAX: ffffffffffffffda RBX: 0000559f81ebce40 RCX: 00007fe73575547a
[Tue Oct  4 10:20:51 2022] RDX: 00007fffa420fa30 RSI: 0000559f84d60da0 RDI: 0000000000000020
[Tue Oct  4 10:20:51 2022] RBP: 0000559f8214b360 R08: 0000000000000100 R09: 0000000000000010
[Tue Oct  4 10:20:51 2022] R10: 0000000000000100 R11: 0000000000000202 R12: 0000559f8214b360
[Tue Oct  4 10:20:51 2022] R13: 00007fffa420fa30 R14: 0000000000000100 R15: 00007fffa420fbe4
[Tue Oct  4 10:20:51 2022]  </TASK>
[Tue Oct  4 10:20:51 2022] Modules linked in: ebtable_filter ebtables ip_set ip6table_raw iptable_raw ip6table_filter ip6_tables iptable_filter sctp ip6_udp_tunnel udp_tunnel nf_tables 8021q garp mrp softdog bonding tls openvswitch nsh nf_conncount nf_nat nf_conntrack nf_defrag_ipv6 nf_defrag_ipv4 nfnetlink_log nfnetlink snd_hda_codec_hdmi snd_hda_codec_conexant snd_hda_codec_generic ledtrig_audio intel_rapl_msr intel_rapl_common intel_tcc_cooling x86_pkg_temp_thermal intel_powerclamp coretemp kvm_intel kvm irqbypass iwlmvm crct10dif_pclmul mei_hdcp ghash_clmulni_intel i915 mac80211 libarc4 aesni_intel crypto_simd cryptd snd_hda_intel drm_buddy rapl intel_cstate btusb snd_intel_dspcfg ttm snd_intel_sdw_acpi iwlwifi btrtl drm_display_helper snd_hda_codec btbcm cec hp_wmi btintel pcspkr platform_profile btmtk cdc_mbim rc_core snd_hda_core cdc_wdm snd_hwdep wmi_bmof sparse_keymap efi_pstore snd_pcm drm_kms_helper i2c_algo_bit ee1004 fb_sys_fops cfg80211 syscopyarea bluetooth snd_timer mei_me cdc_ncm snd
[Tue Oct  4 10:20:51 2022]  sysfillrect cdc_ether ecdh_generic ecc usbnet soundcore sysimgblt mii mei acpi_pad mac_hid vhost_net vhost vhost_iotlb tap ib_iser rdma_cm iw_cm ib_cm ib_core iscsi_tcp libiscsi_tcp libiscsi scsi_transport_iscsi drm sunrpc ip_tables x_tables autofs4 btrfs blake2b_generic simplefb uas usb_storage nvme i2c_i801 crc32_pclmul i2c_smbus e1000e xhci_pci xhci_pci_renesas ahci nvme_core libahci xhci_hcd wmi video
[Tue Oct  4 10:20:51 2022] CR2: 0000000000000016
[Tue Oct  4 10:20:51 2022] ---[ end trace 0000000000000000 ]---
[Tue Oct  4 10:20:52 2022] RIP: 0010:bch2_btree_node_mem_alloc+0x1c2/0x5d0
[Tue Oct  4 10:20:52 2022] Code: 01 00 00 4c 89 ab 50 10 00 00 4c 89 f7 e8 a6 1f bd ff 4c 89 f7 e8 ee 21 bd ff 48 8b 7d d0 e8 85 31 9e 00 49 83 7f 50 00 74 46 <0f> 0b be 00 20 00 00 48 89 df e8 ff e3 ff ff 49 89 c7 48 85 c0 0f
[Tue Oct  4 10:20:52 2022] RSP: 0000:ffffbb1700b075d0 EFLAGS: 00010286
[Tue Oct  4 10:20:52 2022] RAX: ffff939210a34e00 RBX: ffff939207cc0000 RCX: 0000000000000004
[Tue Oct  4 10:20:52 2022] RDX: 0000000000000000 RSI: 0000000000000000 RDI: ffff939207cc1010
[Tue Oct  4 10:20:52 2022] RBP: ffffbb1700b07610 R08: 0000000000000001 R09: 0000000000000000
[Tue Oct  4 10:20:52 2022] R10: 0000000000000004 R11: fffffffffffffff8 R12: ffff939207cc1040
[Tue Oct  4 10:20:52 2022] R13: ffff939204f1d3d0 R14: ffff939204f1d200 R15: ffff9392ffa22c00
[Tue Oct  4 10:20:52 2022] FS:  00007fe7318d6a40(0000) GS:ffff93990f780000(0000) knlGS:0000000000000000
[Tue Oct  4 10:20:52 2022] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[Tue Oct  4 10:20:52 2022] CR2: 0000000000000016 CR3: 00000006d94de003 CR4: 00000000003706e0
[Tue Oct  4 10:20:52 2022] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
[Tue Oct  4 10:20:52 2022] DR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400

```

**Comment by MoraFermi** [comment]
From FSCK log:
```
[Tue Oct  4 10:40:15 2022] fs has wrong cached: got 13433190729560, should be 2827994968, fixing
[Tue Oct  4 10:40:15 2022] fs has wrong cached: 1/1 [3]: got 13431051501856, should be 688767264, fixing
```

```
# bcachefs show-super /dev/sda
External UUID:                  b75126da-9458-418e-9968-6231d8da654c
Internal UUID:                  fdae0d47-e433-493d-865e-da92f78b5299
Device index:                   4
Label:                          DATA
Version:                        backpointers
Oldest version on disk:         backpointers
Created:                        Mon Aug  8 09:15:54 2022
Sequence number:                230
Superblock size:                6072
Clean:                          0
Devices:                        5
Sections:                       members,replicas_v0,disk_groups,clean,journal_seq_blacklist,journal_v2,counters
Features:                       journal_seq_blacklist_v3,new_siphash,inline_data,new_extent_overwrite,btree_ptr_v2,extents_above_btree_updates,btree_updates_journalled,new_varint,journal_no_flush,alloc_v2,extents_across_btree_nodes
Compat features:                alloc_info,alloc_metadata,extents_above_btree_updates_done,bformat_overflow_done

Options:
  block_size:                   4.00 KiB
  btree_node_size:              256 KiB
  errors:                       continue [ro] panic
  metadata_replicas:            2
  data_replicas:                1
  metadata_replicas_required:   1
  data_replicas_required:       1
  encoded_extent_max:           64.0 KiB
  metadata_checksum:            none [crc32c] crc64 xxhash
  data_checksum:                none crc32c [crc64] xxhash
  compression:                  [none] lz4 gzip zstd
  background_compression:       [none] lz4 gzip zstd
  str_hash:                     crc32c crc64 [siphash]
  metadata_target:              none
  foreground_target:            none
  background_target:            none
  promote_target:               none
  erasure_code:                 0
  inodes_32bit:                 1
  shard_inode_numbers:          1
  inodes_use_key_cache:         1
  gc_reserve_percent:           8
  gc_reserve_bytes:             0 B
  root_reserve_percent:         0
  wide_macs:                    0
  acl:                          1
  usrquota:                     1
  grpquota:                     0
  prjquota:                     0
  journal_flush_delay:          1000
  journal_flush_disabled:       0
  journal_reclaim_delay:        100

members (size 288):
  Device:                       0
    UUID:                       90058519-582c-40f9-baba-fe3b3e0f68e4
    Size:                       406 GiB
    Bucket size:                256 KiB
    First bucket:               0
    Buckets:                    1663616
    Last mount:                 Tue Oct  4 10:33:18 2022
    State:                      rw
    Label:                      1 (1)
    Data allowed:               journal,btree,user
    Has data:                   journal,user,cached
    Discard:                    1
    Freespace initialized:      1
  Device:                       1
    UUID:                       582043c6-12d3-465b-abdd-bc97cb2f8069
    Size:                       4.55 TiB
    Bucket size:                512 KiB
    First bucket:               0
    Buckets:                    9538614
    Last mount:                 Tue Oct  4 10:33:18 2022
    State:                      rw
    Label:                      1 (3)
    Data allowed:               journal,btree,user
    Has data:                   journal,btree,user,cached
    Discard:                    1
    Freespace initialized:      1
  Device:                       2
    UUID:                       6f01ef1b-41c8-4e17-bda9-fbec236c1564
    Size:                       4.55 TiB
    Bucket size:                512 KiB
    First bucket:               0
    Buckets:                    9538614
    Last mount:                 Tue Oct  4 10:33:18 2022
    State:                      rw
    Label:                      2 (4)
    Data allowed:               journal,btree,user
    Has data:                   journal,btree,user,cached
    Discard:                    1
    Freespace initialized:      1
  Device:                       3
    UUID:                       8b7a3ded-617c-41d5-8a72-f31527b6b32d
    Size:                       466 GiB
    Bucket size:                512 KiB
    First bucket:               0
    Buckets:                    953880
    Last mount:                 Tue Oct  4 10:33:18 2022
    State:                      rw
    Label:                      2 (5)
    Data allowed:               journal,btree,user
    Has data:                   journal,btree,cached
    Discard:                    0
    Freespace initialized:      1
  Device:                       4
    UUID:                       2d3f0590-806c-4b3f-a368-d75e1c8bc0fa
    Size:                       3.64 TiB
    Bucket size:                1.00 MiB
    First bucket:               0
    Buckets:                    3815447
    Last mount:                 Tue Oct  4 10:33:18 2022
    State:                      rw
    Label:                      3 (6)
    Data allowed:               journal,btree,user
    Has data:                   journal,btree,user,cached
    Discard:                    0
    Freespace initialized:      1
```

```
# bcachefs fs usage -h /srv/
Filesystem: b75126da-9458-418e-9968-6231d8da654c
Size:                       12.5 TiB
Used:                       2.60 TiB
Online reserved:                 0 B

Data type       Required/total  Devices
btree:          1/2             [nvme0n1p3 sdb]             2.50 MiB
btree:          1/2             [sdc sda]                   2.83 GiB
btree:          1/2             [nvme0n1p3 sdc]             12.0 MiB
btree:          1/2             [sdc sdd]                   12.6 GiB
btree:          1/2             [sdd sda]                   2.79 GiB
btree:          1/2             [nvme0n1p3 sdd]             29.0 MiB
btree:          1/2             [nvme0n1p3 sda]             18.5 MiB
btree:          1/2             [sdc sdb]                    347 MiB
btree:          1/2             [sdd sdb]                    318 MiB
btree:          1/2             [sdb sda]                    309 MiB
user:           1/1             [nvme0n1p3]                 6.17 GiB
user:           1/1             [sda]                       5.32 GiB
user:           1/1             [sdd]                       1.12 TiB
user:           1/1             [sdc]                       1.12 TiB
user:           1/1             [sdb]                        170 MiB
user:           1/2             [sdc sdd]                    312 GiB
cached:         1/1             [sdc]                        301 GiB
cached:         1/1             [sdb]                        328 GiB
cached:         1/1             [nvme0n1p3]                  236 GiB
cached:         1/1             [sdd]                        301 GiB
cached:         1/1             [sda]                        182 GiB

hdd.1 (device 1):                sdc              rw
                                data         buckets    fragmented
  free:                          0 B         6215154
  sb:                       3.00 MiB               7       508 KiB
  journal:                  4.00 GiB            8192
  btree:                    7.88 GiB           24895      4.28 GiB
  user:                     1.27 TiB         2674681      19.7 MiB
  cached:                    301 GiB          615685
  parity:                        0 B               0
  stripe:                        0 B               0
  need_gc_gens:                  0 B               0
  need_discard:                  0 B               0
  erasure coded:                 0 B               0
  capacity:                 4.55 TiB         9538614

hdd.2 (device 2):                sdd              rw
                                data         buckets    fragmented
  free:                          0 B         6212787
  sb:                       3.00 MiB               7       508 KiB
  journal:                  4.00 GiB            8192
  btree:                    7.85 GiB           24820      4.26 GiB
  user:                     1.28 TiB         2677119      9.77 MiB
  cached:                    301 GiB          615689
  parity:                        0 B               0
  stripe:                        0 B               0
  need_gc_gens:                  0 B               0
  need_discard:                  0 B               0
  erasure coded:                 0 B               0
  capacity:                 4.55 TiB         9538614

hdd.3 (device 4):                sda              rw
                                data         buckets    fragmented
  free:                          0 B         3609881
  sb:                       3.00 MiB               4      1020 KiB
  journal:                  8.00 GiB            8192
  btree:                    2.97 GiB            5138      2.04 GiB
  user:                     5.32 GiB            5450      5.92 MiB
  cached:                    182 GiB          186782
  parity:                        0 B               0
  stripe:                        0 B               0
  need_gc_gens:                  0 B               0
  need_discard:                  0 B               0
  erasure coded:                 0 B               0
  capacity:                 3.64 TiB         3815447

ssd.1 (device 0):          nvme0n1p3              rw
                                data         buckets    fragmented
  free:                          0 B          661691
  sb:                       3.00 MiB              13       252 KiB
  journal:                  2.00 GiB            8192
  btree:                    31.0 MiB             124
  user:                     6.17 GiB           25262       872 KiB
  cached:                    236 GiB          968334
  parity:                        0 B               0
  stripe:                        0 B               0
  need_gc_gens:                  0 B               0
  need_discard:                  0 B               0
  erasure coded:                 0 B               0
  capacity:                  406 GiB         1663616

ssd.2 (device 3):                sdb              rw
                                data         buckets    fragmented
  free:                          0 B          272081
  sb:                       3.00 MiB               7       508 KiB
  journal:                  3.64 GiB            7452
  btree:                     488 MiB            1368       196 MiB
  user:                      170 MiB             359      9.32 MiB
  cached:                    328 GiB          672613
  parity:                        0 B               0
  stripe:                        0 B               0
  need_gc_gens:                  0 B               0
  need_discard:                  0 B               0
  erasure coded:                 0 B               0
  capacity:                  466 GiB          953880
```

**Comment by koverstreet** [comment]
Fixed by 1ae0d7c505
